contrib/opencensus-ext-azure/opencensus/ext/azure/common/transport.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import requests
+
+from opencensus.trace import execution_context
+
+logger = logging.getLogger(__name__)
+
+
+class TransportMixin(object):
+    def _transmit_from_storage(self):
+        for blob in self.storage.gets():
+            if blob.lease(self.options.timeout + 5):
```

Comment:
Good feedback. I've added comments here.

The purpose is to reduce the chance of blob lease timeout during transmission (`options.timeout` is for networking timeout, so we added a few seconds here). This would reduce the chance where blob is being sent to the network, while lease timed out, and blob is picked up and sent again by another process (which has extra cost and introduce duplicated data).